module/VuFind/src/VuFind/DeveloperSettings/DeveloperSettingsService.php

```diff
@@ -0,0 +1,242 @@
+    protected int $updateInterval = 60;
+
+    /**
+     * Constructor.
+     *
+     * @param ApiKeyServiceInterface $apiKeyService  API key database service
+     * @param array                  $apiKeySettings Section API_Keys from main configuration.
+     */
+    public function __construct(
+        protected ApiKeyServiceInterface $apiKeyService,
+        protected array $apiKeySettings
+    ) {
+        $this->keyLimitPerUser = $apiKeySettings['key_limit'] ?? 5;
+    }
+
+    /**
+     * Generate a new api key token
+     *
+     * @param UserEntityInterface $user User to create salt for
+     *
+     * @return string
+     */
+    protected function createRandomToken(UserEntityInterface $user): string
+    {
+        $salt = $this->apiKeySettings['token_salt'] ?? null;
+        if (!$salt) {
+            throw new \Exception('DeveloperSettingsService: Salt missing');
+        }
+        $valuesForToken = [
+            $user->getEmailVerified()->format('Y-m-d'),
+            $user->getFirstname(),
+            $user->getLastname(),
+            (string)strtotime('now'),
+            $salt,
+        ];
+        return hash('sha256', implode('|', $valuesForToken));
+    }
+
+    /**
+     * Retrieve API keys for user.
+     *
+     * @param UserEntityInterface $user User
+     *
+     * @return ApiKeyEntityInterface[]
+     */
+    public function getApiKeysForUser(UserEntityInterface $user): array
+    {
+        return $this->apiKeyService->getApiKeysForUser($user);
+    }
+
+    /**
+     * Get API key using token.
+     *
+     * @param string $token Token to search for.
+     *
+     * @return ?ApiKeyEntityInterface
+     */
+    public function getApiKeyByToken(string $token): ?ApiKeyEntityInterface
+    {
+        $apiKey = $this->apiKeyService->getByToken($token);
+        if ($apiKey) {
+            $this->updateApiKeyTimeStamp($apiKey);
+        }
+        return $apiKey;
+    }
+
+    /**
+     * Generate an API key for a user.
+     *
+     * @param UserEntityInterface $user  User
+     * @param string              $title Title for the API key
+     *
+     * @return ApiKeyEntityInterface|false API key entity on success, false on failure.
+     */
+    public function generateApiKeyForUser(UserEntityInterface $user, string $title): ApiKeyEntityInterface|false
+    {
+        $tokens = $this->apiKeyService->getApiKeysForUser($user);
+        if ($this->apiKeysBlocked($tokens)) {
+            return false;
+        }
+        // Generate unique id from date and users id.
+        $newKey = $this->apiKeyService->createEntity();
+        $newKey->setToken($this->createRandomToken($user))
+            ->setUser($user)
+            ->setCreated(new DateTime())
+            ->setLastUsed(new DateTime())
+            ->setTitle($title);
+        $this->apiKeyService->persistEntity($newKey);
+        return $newKey;
+    }
+
+    /**
+     * Set the last used value to the API key. At default this will be only updated once every hour.
```

Comment:
Minor wording suggestion:

```suggestion
     * Set the last used value to the API key. By default this will be only updated once every hour.
```